docs/excel/custom-functions-best-practices.md

```diff
@@ -0,0 +1,184 @@
+---
+ms.date: 09/18/2018
+description: Learn best practices and recommended patterns for Excel custom functions.
+title: Custom Functions' best practices
+---
+
+# Custom Functions' best practices
+
+This article covers some recommended patterns and solutions to common use cases with Excel custom functions.
+
+## Authentication
+
+Non-custom functions add-ins which have web views (i.e. display information in the taskpane) accomplish authentication via `Office.context.displayDialogAsync`. Custom functions differ from this approach. Because custom functions use a different runtime, they do not have access to the Office.context methods. Instead, they use methods available on the `OfficeRuntime` object, such as `OfficeRuntime.DisplayWebDialog()` and `OfficeRuntime.AsyncStorage.setItem(key, value)`.
+
+The following code sample shows how to store a token in AsyncStorage and display a dialog box which can indicate whether or not the user is authenticated.  
+
+```js
+// Get auth token before calling my service, a hypothetical API, which will deliver a stock price based on stock ticker string, such as "MSFT"
+async function getStock(ticker) {
+    const token = await getToken();
+    let data = await (await fetch(https://myservice.com/?token=token&ticker= + ticker).json());
+    return data.price;
+}
+
+async function getToken() {
+    if (_cachedToken) {
+        return _cachedToken;
+    } else {
+        return await getTokenViaDialog_AsPromise();
+    }
+}
+  
+// Function to display dialog window
+function getTokenViaDialog_AsPromise() {
+    return new Promise ((resolve, reject) => {
+        displayWebDialog("https://www.auth.com/", {
+           height: ’50’,
```

Comment:
width has a % sign in its value, but height does not. Which is correct? Another reason to suspect that this code was not tested.

Reply:
I talked with Michael and Kevin Shi on this and it sounds like percentage it the correct one. Should be reflected in the new sample.